specs/language/basic.tex

```diff
@@ -1,5 +1,97 @@
+\textit{N} equals \texttt{sizeof(T)}\footnote{\texttt{sizeof(T)}, returns the
+size of an object of type \texttt{T} in bytes as stored in device memory.
+The size may be different in other memory spaces and no operator exists to query
+it.}. The \textit{object representation} of an object may be different based on
+the \textit{memory space} it is stored in (\ref{Intro.Memory.Spaces}).
+
+\p The \textit{value representation} of an object is the set of bits that hold
+the value of type \texttt{T}.
+
+\p An \textit{object type} \texttt{T} is an \textit{incomplete type} if the
+compiler lacks sufficient information to determine the size of an object of type
+\texttt{T}, and a \textit{complete type} if the compiler has sufficient
+information to determine the size of an object of type \texttt{T}. An object may
+not be defined to have an \textit{incomplete} type.
+
+\p A \textit{class type} is a data type declared with either the \texttt{class}
+or \texttt{struct} keywords (\ref{Classes}). A class type \texttt{T} may be
+declared as incomplete at one point in a translation unit via a \textit{forward
+declaration}, and complete later with a full definition. The type \texttt{T} is
+the same type throughout the translation unit.
+
+\p An \textit{object type} is a type that is not a function type, not a
+reference type, and not a void type.
+
+\p Arithmetic types (\ref{Basic.types.arithmetic}), enumeration types, and
+\textit{cv-qualified} versions of these types are collectively called
+\textit{scalar types}.
+
+\p Vectors of scalar types declared with the built-in \texttt{vector<T,N>}
+template are \textit{vector types}.
+
+\p \textit{POD types}\footnote{POD is an initialism for Plain-Old-Data which is
+referenced in the C++ specification but never expanded.} are POD class types,
+scalar types, vector types, and arrays of such types. \textit{POD class types}
+are class types that only contain members POD types.
+
+\Sub{Arithmetic Types}{Basic.types.arithmetic}
+
+\p There are three \textit{standard signed integer types}: \texttt{int16\_t},
+\texttt{int32\_t}, and \texttt{int64\_t}. Each of the signed integer types is
+explicitly named for the size in bits of the type's object representation. There
+is also the type alias \texttt{int} which is an alias of \texttt{int32\_t}. The
+standard signed integer types are collectively called \textit{signed integer
+types}.
+
+\p There are three \textit{standard unsigned integer types}: \texttt{uint16\_t},
+\texttt{uint32\_t}, and \texttt{uint64\_t}. Each of the unsigned integer types is
+explicitly named for the size in bits of the type's object representation. There
+is also the type alias \texttt{uint} which is an alias of \texttt{uint32\_t}. The
+standard unsigned integer types are collectively called \textit{unsigned integer
+types}\footnote{HLSL does not inherit C's default type rule, nor does it
+inherit the \texttt{unsigned} keyword.}.
```

Comment:
I find "C's default type rule" fairly ambiguous - C has a lot of defaults when it comes to types. Is this simply saying that HLSL does not have C's `signed` and `unsigned` modifiers for integral types?